### PR TITLE
Fix curl cycle

### PR DESCRIPTION
**Pull request by ocefpaf**
@nehaljwani can you take a quick look?

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
   build:
-    - python         # [win]
-    - cmake >2.8
+    - python  # [win]
+    # This breals a dependency cycle:
+    # curl->libssh2->cmake->curl
+    - cmake-binary
     - openssl 1.0.*
     - toolchain
+    - zlib zlib >=1.2.11,<1.3.0a0
```
Comment (nehaljwani) [inline recipe/meta.yaml]: Whoops.

Comment (ocefpaf) [reply recipe/meta.yaml]: Fixed :stuck_out_tongue_winking_eye:

**Comment by ocefpaf** [comment]
@nehaljwani locally I got `/opt/conda/conda-bld/libssh2_1509717534394/work/libssh2-1.8.0/conda_build.sh: line 6: cmake: command not found` :unamused: 

PS: Why did you avoid the `zlib` in the `defaults` version? Should we do the same here?

File: recipe/build.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# We use a repackaged cmake from elsewhere to break a build cycle.
+export PATH=${PREFIX}/cmake-bin/bin:${PATH}
```
Comment (ocefpaf) [inline recipe/build.sh:4]: @nehaljwani I got this from your recipe and now ti works, but `ctest` fails.
I see that you are not running them, any advice?

Comment (nehaljwani) [reply recipe/build.sh:4]: For posterity sake, the build that failed was: https://circleci.com/gh/conda-forge/libssh2-feedstock/24
I'll look into why the ctest fails inside docker container this weekend.

I didn't modify the libssh2-feedstock in AnacondaRecipes. That was @mingwandroid 's work.

Comment (ocefpaf) [reply recipe/build.sh:4]: Thanks for the info @nehaljwani!

**Pull request merged**